src/dom-wrapper.ts

```diff
@@ -78,6 +79,52 @@ export class DOMWrapper<ElementType extends Element> implements WrapperAPI {
     return this.trigger('change')
   }
 
+  setValue(value?: any) {
+    const element = (this.element as unknown) as HTMLInputElement
+    const tagName = element.tagName
+    const type = this.attributes().type
+
+    if (tagName === 'OPTION') {
+      return this.setSelected()
+    } else if (tagName === 'INPUT' && type === 'checkbox') {
+      return this.setChecked(value)
+    } else if (tagName === 'INPUT' && type === 'radio') {
+      return this.setChecked(value)
+    } else if (
+      tagName === 'INPUT' ||
+      tagName === 'TEXTAREA' ||
+      tagName === 'SELECT'
+    ) {
+      element.value = value
+
+      if (tagName === 'SELECT') {
+        return this.trigger('change')
+      }
+      this.trigger('input')
+      // trigger `change` for `v-model.lazy`
+      return this.trigger('change')
```

Comment:
Had to rewrite specs for an input featuring the reward-early-punish-late validation strategy. Not a big problem, but it would be nice to mention this in the breaking changes.

Reply:
Hello from 2022 (almost 2023)!
This can indeed be problematic when it comes to asserting form validation strategies. An example is known as [rewardEarlyPunishLate](https://vuelidate-next.netlify.app/examples.html#returning-metadata-from-validators).  Simplifying the problem and is more or less like this: 
1. You have an invalid input with an error message displayed; 
1. typing on that input (`setValue`) should remove the validation (**rewardEarly**) 
1. if the input is still invalid after typing, it only re-validates on change (blur/keydown.enter) (**punishLate**)

```js
// pseudo code
it('removes validation message when input using rewardEarlyPunishLate validation mode') {
  const wrapper = mount(InputComp, { props: { modelValue: 'foo', rules: [mustBeFooBar] } })
  const input = wrapper.find('input')
  const validation = wrapper.find('[data-test-id="validation-container"]')
 
  expect(validation.text()).toBe('must be FooBar')
  await input.setValue('foo-b')
  // it rewards early so it should remove validation message as soon as user types
  expect(validation.text()).toEqual('') //  but now it breaks, since it triggers `change` event it validates again and renders the "must be FooBar"
}
```

**Current Alternative:** manually setting `input.element.value` and `trigger('input')`. 

```diff
// pseudo code
it('removes validation message when input using rewardEarlyPunishLate validation mode') {
  const wrapper = mount(InputComp, { modelValue: 'foo', rules: [mustBeFooBar] })
  
  const validation = wrapper.find('[data-test-id="validation-container"]')
 
  expect(validation.text()).toBe('must be FooBar')
-  await input.setValue('foo-b')
+ input.element.value = 'foo-b'
+ await input.trigger('input')
  // it works
  expect(validation.text()).toEqual('') 
}
```

I guess it's ok, not sure if this con justifies a fix, but maybe docs could mention this somewhere.

Cheers! 👋